packages/vm/src/imports.rs

```diff
@@ -949,6 +932,23 @@ mod test {
         assert_eq!(item.unwrap().unwrap().0, (KEY1.to_vec(), VALUE1.to_vec()));
     }
 
+    #[test]
+    #[cfg(feature = "iterator")]
+    fn do_scan_errors_for_invalid_order_value() {
+        let mut instance = make_instance();
+        let ctx = instance.context_mut();
+        leave_default_data(ctx);
+
+        // set up iterator over all space
+        let result = do_scan::<MS, MQ>(ctx, 0, 0, 42);
+        match result.unwrap_err() {
+            VmError::CommunicationErr {
+                source: CommunicationError::InvalidOrder { .. },
```

Comment:
If you add the invalid order value, then add a check here that it has the correct value.

Reply:
Done in 84e991c